client/client.go

```diff
@@ -145,54 +154,68 @@ func redactPasswordArgs(args []string) []string {
+		Email:          email,
+		PasswordBase64: base64.StdEncoding.EncodeToString([]byte(password)),
+	})
+	if w.conf.Timeout > 0 {
+		params.SetTimeout(w.conf.Timeout)
+	}
+	if p != nil {
+		if p.Timeout > 0 {
+			params.SetTimeout(p.Timeout)
+		}
+		params.SetXGiantSwarmActivity(&p.ActivityName)
+		params.SetXGiantSwarmCmdLine(&p.CommandLine)
+		params.SetXRequestID(&p.RequestID)
+	}
 
-func (c *ClientWrapper) GetCluster(clusterID string, activityName string) (*gsclientgen.V4ClusterDetailsModel, *gsclientgen.APIResponse, error) {
-	response, apiResponse, err := c.client.GetCluster(c.authHeader, clusterID, c.requestID, activityName, getCommandLine())
-	return response, apiResponse, err
-}
-func (c *ClientWrapper) GetClusters(activityName string) ([]gsclientgen.V4ClusterListItem, *gsclientgen.APIResponse, error) {
-	response, apiResponse, err := c.client.GetClusters(c.authHeader, c.requestID, activityName, getCommandLine())
-	return response, apiResponse, err
-}
-func (c *ClientWrapper) GetInfo(activityName string) (*gsclientgen.V4InfoResponse, *gsclientgen.APIResponse, error) {
-	response, apiResponse, err := c.client.GetInfo(c.authHeader, c.requestID, activityName, getCommandLine())
-	return response, apiResponse, err
-}
-func (c *ClientWrapper) GetKeyPairs(clusterID string, activityName string) ([]gsclientgen.KeyPairModel, *gsclientgen.APIResponse, error) {
-	response, apiResponse, err := c.client.GetKeyPairs(c.authHeader, clusterID, c.requestID, activityName, getCommandLine())
-	return response, apiResponse, err
-}
-func (c *ClientWrapper) GetReleases(activityName string) ([]gsclientgen.V4ReleaseListItem, *gsclientgen.APIResponse, error) {
-	response, apiResponse, err := c.client.GetReleases(c.authHeader, c.requestID, activityName, getCommandLine())
-	return response, apiResponse, err
-}
-func (c *ClientWrapper) GetUserOrganizations(activityName string) ([]gsclientgen.V4OrganizationListItem, *gsclientgen.APIResponse, error) {
-	response, apiResponse, err := c.client.GetUserOrganizations(c.authHeader, c.requestID, activityName, getCommandLine())
-	return response, apiResponse, err
-}
-func (c *ClientWrapper) ModifyCluster(clusterID string, body gsclientgen.V4ModifyClusterRequest, activityName string) (*gsclientgen.V4ClusterDetailsModel, *gsclientgen.APIResponse, error) {
-	response, apiResponse, err := c.client.ModifyCluster(c.authHeader, clusterID, body, c.requestID, activityName, getCommandLine())
-	return response, apiResponse, err
-}
-func (c *ClientWrapper) UserLogin(email string, body gsclientgen.LoginBodyModel, activityname string) (*gsclientgen.LoginResponseModel, *gsclientgen.APIResponse, error) {
-	response, apiResponse, err := c.client.UserLogin(email, body, c.requestID, activityname, cmdLine)
-	return response, apiResponse, err
+	response, err := w.gsclient.AuthTokens.CreateAuthToken(params, nil)
+	if err != nil {
+		return nil, clienterror.New(err)
+	}
+
+	return response.Payload, nil
```

Comment:
Why not returning the whole response here?
I feel it could be helpful in the future to be able to access the whole response object

Reply:
Changed to returning the whole response

Reply:
The thinking is here, is to provide as access to as much information as we can to the caller function and not having to change that in the future, because caller need access to `response.Something`.